diff --git a/src/libprojectM/projectM-opengl.h b/src/libprojectM/projectM-opengl.h
index 8a605ca..b8056c8 100644
--- a/src/libprojectM/projectM-opengl.h
+++ b/src/libprojectM/projectM-opengl.h
@@ -11,8 +11,17 @@
 #define UNLOCK_FPS 0
 
 #ifdef __APPLE__
-# include <OpenGL/gl3.h>
-# include <OpenGL/gl3ext.h>
+# include <TargetConditionals.h>
+# if TARGET_OS_IPHONE || defined(USE_GLES)
+#  include <OpenGLES/ES3/gl.h>
+#  include <OpenGLES/ES3/glext.h>
+#  ifndef GL_APIENTRY
+#   define GL_APIENTRY
+#  endif
+# else
+#  include <OpenGL/gl3.h>
+#  include <OpenGL/gl3ext.h>
+# endif
 #elif defined(EYETUNE_WINRT)
 # define GL_GLEXT_PROTOTYPES
 # define GLM_FORCE_CXX03
